docs/contributing/local-testing.md

```diff
@@ -0,0 +1,184 @@
+  oci:
+    - name: oci-1
+      privateKey: |
+        -----BEGIN RSA PRIVATE KEY-----
+        ..... put the private key here ....
+        -----END RSA PRIVATE KEY-----
+      keyFingerprint: "key fingerprint"
+      tenancyOcid: "tenancy ocid"
+      userOcid: "user ocid"
+      compartmentOcid: "compartment ocid"
+  aws:
+    - name: aws-1
+      accessKey: "access key"
+      secretKey: "secret key"
+  azure:
+    - name: azure-1
+      subscriptionId: "subscription id"
+      tenantId: "tenant id"
+      clientId: "client id"
+      clientSecret: "client secret"
+  hetznerdns:
+    - name: hetznerdns-1
+      apiToken: "api token"
+  cloudflare:
+    - name: cloudflare-1
+      apiToken: "api token"
+
+nodePools:
+  dynamic:
+    - name: hetzner-control
+      providerSpec:
+        name: hetzner-1
+        region: nbg1
+        zone: nbg1-dc3
+      count: 1
+      serverType: cpx11
+      image: ubuntu-22.04
+
+    - name: hetzner-compute
+      providerSpec:
+        name: hetzner-1
+        region: nbg1
+        zone: nbg1-dc3
+      count: 1
+      serverType: cpx11
+      image: ubuntu-22.04
+      storageDiskSize: 50
+
+    - name: hetzner-lb
+      providerSpec:
+        name: hetzner-1
+        region: nbg1
+        zone: nbg1-dc3
+      count: 1
+      serverType: cpx11
+      image: ubuntu-22.04
+
+  static:
+    - name: static-pool
+      nodes:
+        - endpoint: "192.168.52.1"
+          privateKey: |
+            -----BEGIN RSA PRIVATE KEY-----
+            ...... put the private key here .....
+            -----END RSA PRIVATE KEY-----
+        - endpoint: "192.168.52.2"
+          privateKey: |
+            -----BEGIN RSA PRIVATE KEY-----
+            ...... put the private key here .....
+            -----END RSA PRIVATE KEY-----
+
+kubernetes:
+  clusters:
+    - name: dev-test
+      version: v1.26.1
+      network: 192.168.2.0/24
+      pools:
+        control:
+          - static-pool
+        compute:
+          - hetzner-compute
+
+loadBalancers:
+  roles:
+    - name: apiserver-lb
+      protocol: tcp
+      port: 6443
+      targetPort: 6443
+      target: k8sControlPlane
+  clusters:
+    - name: miro-lb
+      roles:
+        - apiserver-lb
+      dns:
+        dnsZone: zone.com
+        provider: cloudflare-1
+      targetedK8s: dev-test
+      pools:
+        - hetzner-lb
+```
+
+### Test Claudie using manual manifest injection
+
+To test Claudie in a more "manual" way, you can use the test client to inject an input manifest. The code for the client can be found in `services/context-box/client/client_test.go`, specifically in the `TestSaveConfigFrontEnd` function.
+
+In this function, the input manifest (in raw YAML format, not CRD) is located based on the `manifestFile` variable and applied to Claudie. It's important to note that this method of testing does not provide automatic clean up or verification of Longhorn deployment. Therefore, exercise caution when using this testing approach.
```

Comment:
while you can push a config with this, you can't delete.

Actually is pretty simple to just write a dead-simple grpc Client (which I've been using anyways) with also the validation (without providers)

```go
func main() {
	file, err := os.ReadFile("./test-manifests/manifest.yml")
	if err != nil {
		panic(err)
	}

	var m manifest.Manifest
	if err := yaml.Unmarshal(file, &m); err != nil {
		panic(err)
	}

	if err := m.Validate(); err != nil {
		panic(err)
	}

	conn, err := grpc.Dial("localhost:50055", grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		panic(err)
	}

	cbox := pb.NewContextBoxServiceClient(conn)
	createConfig(cbox, &m, file)
}

func deleteConfg(cbox pb.ContextBoxServiceClient, m *manifest.Manifest) {
	_, err := cbox.DeleteConfig(context.Background(), &pb.DeleteConfigRequest{
		Id:   m.Name,
		Type: pb.IdType_NAME,
	})
	if err != nil {
		panic(err)
	}
}

func createConfig(cbox pb.ContextBoxServiceClient, m *manifest.Manifest, rawManifest []byte) {
	config := &pb.Config{
		Name:     m.Name,
		Manifest: string(rawManifest),
	}

	_, err := cbox.SaveConfigFrontEnd(context.Background(), &pb.SaveConfigRequest{Config: config})
	if err != nil {
		panic(err)
	}
}

```

Reply:
I don't doubt you can do something like this with minimal effort. I was just describing what we have right now in the repository. 

There are at least 5 different ways I was testing Claudie over the years, so I just limited the docs to what we have. Also, regarding the deletion, we have the function `TestDeleteConfig` in the same client. 

(Disclaimer: I almost exclusively test stuff with a testing framework to validate Longhorn as well)